packages/feel-input-component/src/FeelInput.tsx

```diff
@@ -105,12 +140,60 @@ export const FeelInput = React.forwardRef<FeelInputRef, FeelInputProps>(
           };
         },
       });
-
       return () => {
         disposable.dispose();
       };
     }, [enabled, suggestionProvider]);
 
+    useEffect(() => {
+      if (!enabled) {
+        return;
+      }
+
+      const disposable = Monaco.languages.registerDocumentSemanticTokensProvider(
+        { language: MONACO_FEEL_LANGUAGE },
+        {
+          provideDocumentSemanticTokens: function (model) {
+            const tokens = new Array<number>();
+
+            if (feelVariables) {
+              const text = model.getValue();
+              const feels = feelVariables.parser.parse(expressionId ?? "", text);
+
+              let lastPosition = 0;
+              let offset = 0;
+              for (const variable of feels) {
+                lastPosition = variable.startIndex;
+                // lineIndex
+                // columnIndex (it's relative to the PREVIOUS token NOT to the start of the line)
+                // tokenLength
+                // token type
+                // token modifier
+                tokens.push(0, lastPosition - offset, variable.length, getTokenTypeIndex(variable.variableType), 0);
+                offset = lastPosition;
+              }
```

Comment:
This comment is supposed to tell the order of the parameters, right? Why not:

```suggestion
              for (const variable of feels) {
                lastPosition = variable.startIndex;
                tokens.push(
                  0, // lineIndex
                   lastPosition - offset, // columnIndex (it's relative to the PREVIOUS token NOT to the start of the line)
                   variable.length, // tokenLength
                   getTokenTypeIndex(variable.variableType), // token type
                   0 // token modifier
                 );
                offset = lastPosition;
              }
```